### ISSUE TITLE
Graph doesn't render in Atom

### ISSUE THREAD

**Issue by stevencollinscn** [kylebarron/stata_kernel#125]
A histgrom displayed in Stata's Graph windows, but Atom returns an error
![snipaste_2018-09-02_13-40-31](https://user-images.githubusercontent.com/33796896/44958995-fe167b80-aeb5-11e8-83e5-50d233a820fc.png)

I also encontered the following issue from time to time and have to reopen Stata 15 to slove it.
![snipaste_2018-09-02_13-38-12](https://user-images.githubusercontent.com/33796896/44958996-fe167b80-aeb5-11e8-9247-dfb58fdb3b98.png)

**Comment by stevencollinscn** [comment]
@kylebarron I'm mostly using Python now and haven't used Stata for a while, and discovered your stata_kernel while trying to set up stata-exac in Atom. I installed the kernel last night and that was the first dataset I tried. 
I can share it here: 
[stata_test.zip](https://github.com/kylebarron/stata_kernel/files/2343771/stata_test.zip)

**Comment by kylebarron** [comment]
I can't reproduce the second error on Linux. You're on Windows right? Do you ever resize the Stata window? With current behavior, if you manually resize the Stata window, the length of Stata output lines may decrease, and this package won't correctly remove the code lines. 

Eventually (maybe soon) this package will completely hide the Stata window on Windows, and add a `debug` flag to the configuration to show the Stata window.

**Comment by stevencollinscn** [comment]
Yes, I'm running Stata on Wins 10.  Yes, I resize Stata every time. If I minimize the window, it will keep flashing out in the taskbar. So... I think it would be a great idea to hide the Stata window and especially the graph windows since it just keep showing up everytime when you create a graph in Stata
<img width="422" alt="snipaste_2018-09-04_17-33-29" src="https://user-images.githubusercontent.com/33796896/45059320-9acd4a80-b069-11e8-973b-ff368bdfa2d7.png">

Btw, anyway the kernel can be indexed in conda or something so that it will be updated via Anaconda

**Comment by kylebarron** [comment]
Yes I'll change it to hide it. I don't know what you mean with your last sentence

**Comment by stevencollinscn** [comment]
I think I'm trying to say that if I can conda install the kernel instead of using pip.

**Comment by kylebarron** [comment]
Eventually maybe but not now.

**Comment by stevencollinscn** [comment]
Just tried to run some `hist `in the jupyter lab. The main Stata window is indeed hidden, but graph windows still appear whenever a graph is generated. Is this something planned to do in the future?

**Comment by stevencollinscn** [comment]
I also wonder if you would be interested in add a magic command like `%labels ` to check variable label since the main window no longer lanunch now. If not, I just gotta use `describe ` for that.

**Comment by kylebarron** [comment]
1. I can't prevent graph windows from opening. Stata is incapable of exporting a graph without creating a window for it. So if I hid the graph window, you wouldn't see a graph in Jupyter.
2. Now that I think about it, I could do a `graph close` _after_ exporting each graph. The window will still flash, but it won't remain open. Would that be an improvement?
3. Originally I wanted the labels to show during autocomplete, but I don't think I can do that. What would you want `%labels` to do that `describe` doesn't?

**Comment by kylebarron** [comment]
I could add a setting for whether the Stata window is hidden or not.

**Comment by stevencollinscn** [comment]
If you can't, I think it is fine to leave the graph windows that way since it won't flash out in the taskbar like the main window does. I actually tried `graph close` but as you said a graph window will still appear. If you run a couple of graphs in a row, Stata will automatically close them and only leave the last one open, so your taskbar won't be full of graph windows. Thus, I'm fine with the way it is now. 

In terms of `%labels`, I think I just want it to show the variable names and their labels only. `describe` provides too much information. It's like you want to check `df.dtypes` but use `df.info()` to do so. I also realized that `%browse` only shows a limited number of rows in the dataset now, so I don't know if it would also be a challenge to show all of the variable labels in jupyter lab or Atom. Stata users tend to put a lot of variables in a single dataset due to the limitation of Stata's data structure. Meanwhile, once you get a lot variables, you may need a search feature to locate variable labels. I googled it, and it doesn't seem Stata has a line commond for that. 

I think it is a good idea to add a setting for Stata's main window in case that someone needs it for certain things e.g., import `.csv`. 

I was actually about to ask you if you are working on a autocomplete for Stata since you already have a package for `stata_language `.

**Comment by kylebarron** [comment]
> %browse only shows a limited number of rows in the dataset now

I could add a setting to browse the entire dataset, but whenever you do `%browse`, you export the data from Stata to Python, which can take a while for large data. That's why there are defaults of a few hundred obs.

> I googled it, and it doesn't seem Stata has a line commond for that.

I actually discovered this today: https://www.stata.com/help.cgi?lookfor

> autocomplete for Stata

This package provides autocompletions based on macros/variables/matrices/scalars. `language-stata` provides autocompletions based on names of commands. I have no plans to add anything more than that because it could be very time consuming.

**Comment by stevencollinscn** [comment]
Thanks man. `lookfor ` seems working very well. Just curious if you are doing anything interesting about Python.

**Comment by kylebarron** [comment]
Like what?

**Comment by stevencollinscn** [comment]
anything about data analysis I guess?

**Comment by kylebarron** [comment]
Anything in Python would need to copy the data and would be slow. Hence I don't plan anything other than just browse.
